### PR TITLE
 #162725269: Implement search and custom filtering functionality to articles

### PR DESCRIPTION
**Pull request by mariamiah**
### What does this PR do?
This PR adds search and custom filtering for the articles

### Description of Task to be completed?
Looking for articles in their dozens can be daunting. Filters shall avail the members to create a search criterion of their own which helps them to narrow it down if they don’t know the title exactly.

### How should this be manually tested?
- fetch this branch by running the command `git fetch origin ft-search-custom-filters-162725269` in the terminal.
- check into the branch `git checkout ft-search-custom-filters-162725269`
- start the virtual environment `source venv/bin/activate`
- re-install requirements `pip install -r requirements.txt`
- start the application `python manage.py runserver`
- Filter the articles  by providing the title and author  
- To filter according to title, use `api/articles/?title=title_name`
- To filter according to author,  use `api/articles/?author=author_name`
- To filter by tags, use `api/articles/?tags=tag_name`
- To filter by favorited, use `api/articles/?favorited=True` or `api/articles/?favorited=False`
- To search for a value within the article use `api/articles/?search=value_name`

What are the relevant pivotal tracker stories?
[#162725269](https://www.pivotaltracker.com/n/projects/2232225/stories/162725269)

### Postman collection
[Run in Postman](https://app.getpostman.com/run-collection/5f5443eb5d0795637193)

### CODE REVIEW

**Comment by malfahad** [comment]
Can you please Include the supported `field_names` in the PR description.

**Comment by MuhweziDeo** [comment]
@mariamiah  the git fetch method is not well highlighted

**Review by MuhweziDeo — DISMISSED** [review]
I approve funtionality for this PR

File: authors/apps/articles/filters.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from django_filters import FilterSet
+from django_filters import rest_framework as filters
+from .models import Article
+
+
+class ArticleFilter(FilterSet):
+    title = filters.CharFilter(lookup_expr='icontains')
+    author = filters.CharFilter('author__user__username',
```
Comment (ja-odur) [inline authors/apps/articles/filters.py:8]: @mariamiah use `icontains` for look_up as well

Comment (ja-odur) [reply authors/apps/articles/filters.py:8]: @mariamiah Use `icontains` for all search fields

Comment (armstrongsouljah) [reply authors/apps/articles/filters.py:8]: @mariamiah Don't forget to squash your commits.

**Review by armstrongsouljah — DISMISSED** [review]
LGTM

**Review by MuhweziDeo — DISMISSED** [review]
Looks good to me

**Review by kdaglas — DISMISSED** [review]
I have tested it, and it works well, its good to go

**Review by ja-odur — APPROVED** [review]
LGTM

**Review by kdaglas — APPROVED** [review]
lgtm

**Review by MuhweziDeo — APPROVED** [review]
Looks good to me

**Review by malfahad — APPROVED** [review]

**Pull request merged**